### DISCUSSION TITLE
angular, set app version on publish

### DISCUSSION THREAD

**Discussion by leon** [nrwl/nx#7783] [General]
We have multiple angular apps in our repo.

I would like to be able to show a version and git commit sha in the app, so we know which one is deployed to production.

## Possible solutions

after the app is build, create a custom executor/script that updates the index.html with a `<script>window.APP_VERSION=${version}`

Is there anyone out there that has solved this or knows of a tutorial?

**Comment by miluoshi** [comment]
We do it by interpolating environment variables after `build in a custom target `postbuild-prepare-index` using `@nrwl/workspace:run-commands` executor. Note that `index.html` file outputted from `build` target in `dist/apps/my-app` is being updated.

`index.html`:
```html
<script>
window.APP_CONFIG = {
  branch: '${GIT_BRANCH}',
  VERSION: '${VERSION}',
}
</script>
```

`workspace.json`:
```json
{
  "projects": {
    "my-app": {
      "targets": {
        "postbuild-prepare-index": {
          "executor": "@nrwl/workspace:run-commands",
          "options": {
            "commands": [
              "mv dist/apps/my-app/index.html dist/apps/my-app/index.html.tmpl",
              "sed -i \"s|\\${GIT_BRANCH}|$GIT_BRANCH|\" dist/apps/my-app/index.html.tmpl",
              "sed -i \"s|\\${VERSION}|$VERSION_ONLY|\" dist/apps/my-app/index.html.tmpl"
            ],
            "parallel": false
          }
        }
      }
    }
  }
}
```

**Reply by leon** [reply]
Just what I was looking for!
Thanks for sharing 👍🏻
